### PR TITLE
CB-2888 project sync

### PR DESCRIPTION
**Pull request by yagudin10**

### CODE REVIEW

File: server/bundles/io.cloudbeaver.server/schema/service.events.graphqls [hunk 1]
```diff
@@ -16,9 +16,13 @@ enum CBServerEventId {
...
 
-    cb_rm_resource_created
-    cb_rm_resource_updated
-    cb_rm_resource_deleted
+    cb_rm_resource_created,
+    cb_rm_resource_updated,
+    cb_rm_resource_deleted,
+
+    cb_rm_project_created,
+    cb_rm_project_updated,
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.server/schema/service.events.graphqls]:
```suggestion
```
remove non-existent event

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSPermissionUpdatedEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        }
+        var permissionEvent = (WSObjectPermissionEvent) event;
+        if (!SMObjects.PROJECT_OBJECT_TYPE_ID.equals(permissionEvent.getSmObjectType().getObjectType())) {
+            return;
+        }
+        var smController = CBPlatform.getInstance().getApplication().getSecurityController();
+        try {
+            var projectId = permissionEvent.getObjectId();
+            var permissionGrants = smController.getObjectPermissionGrants(projectId, permissionEvent.getSmObjectType());
+            var newProjectSubjects = permissionGrants.stream().map(SMObjectPermissionsGrant::getSubjectId).collect(Collectors.toSet());
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSPermissionUpdatedEventHandler.java]:
```suggestion
            var subjectsWithProjectAccess = permissionGrants.stream().map(SMObjectPermissionsGrant::getSubjectId).collect(Collectors.toSet());
```

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java [hunk 3]
```diff
@@ -1,53 +1,49 @@
-/*
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java]: copyright

File: server/bundles/io.cloudbeaver.server/schema/service.events.graphqls [hunk 4]
```diff
@@ -16,9 +16,13 @@ enum CBServerEventId {
...
     cb_datasource_folder_deleted,
 
-    cb_rm_resource_created
-    cb_rm_resource_updated
-    cb_rm_resource_deleted
+    cb_rm_resource_created,
+    cb_rm_resource_updated,
+    cb_rm_resource_deleted,
+
+    cb_rm_project_created,
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.server/schema/service.events.graphqls]: may be it better  this event as  `cb_rm_project_added`, 
in the case of changing access rights, we add this project to the user, and do not create a new

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java [hunk 5]
```diff
@@ -1,53 +1,49 @@
...
-        for (var activeUserSession : allSessions) {
-            if (WSWebUtils.isSessionIdEquals(activeUserSession, event.getSessionId())) {
-                continue;
+    protected Class<WSProjectUpdateEvent> getEventClass() {
+        return WSProjectUpdateEvent.class;
+    }
+
+    @Override
+    protected void updateSessionData(BaseWebSession activeUserSession, WSProjectUpdateEvent event) {
+        if (!activeUserSession.getUserContext().hasPermission(DBWConstants.PERMISSION_ADMIN)) {
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java]: check admin permissions only for RM_PROJECT_CREATED event, when a project is deleted - it must be deleted for all users

File: server/bundles/io.cloudbeaver.server/schema/service.events.graphqls [hunk 6]
```diff
@@ -95,6 +95,13 @@ type CBTopicEvent implements CBClientEvent {
     topicId: CBEventTopic! # topic to subscribe / unsubscribe
 }
 
+# Project event
+type CBProjectEvent implements CBClientEvent {
+    id: CBClientEventId!
+    topicId: CBEventTopic
+    projectId: String!
+}
```
Comment (Wroud) [inline server/bundles/io.cloudbeaver.server/schema/service.events.graphqls:110]: `CBClientEvent` sent by web application to the server
please use `implements CBServerEvent` and `id: CBServerEventId!`

**Review by Wroud — CHANGES_REQUESTED** [review]

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java (hunk 5 above)
Comment (EmYar) [reply server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java]: Fixed

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java (hunk 3 above)
Comment (EmYar) [reply server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java]: Fixed

**Review by Wroud — APPROVED** [review]

File: webapp/packages/core-projects/package.json [hunk 7]
```diff
@@ -24,7 +24,8 @@
     "@cloudbeaver/core-root": "~0.1.0",
     "@cloudbeaver/core-ui": "~0.1.0",
     "@cloudbeaver/core-executor": "~0.1.0",
-    "@cloudbeaver/core-utils": "~0.1.0"
+    "@cloudbeaver/core-utils": "~0.1.0",
+    "@cloudbeaver/core-navigation-tree": "~0.1.0"
```
Comment (Wroud) [inline webapp/packages/core-projects/package.json]: core-navigation-tree can't be used in core-projects (circular dependency)

File: webapp/packages/core-projects/src/ProjectInfoEventHandler.ts [hunk 8]
```diff
@@ -7,15 +7,15 @@
...
 
 import { injectable } from '@cloudbeaver/core-di';
-import { SessionEventSource, TopicEventHandler, ISessionEvent, ClientEventId, SessionEventTopic, IBaseServerEvent, SessionEventId } from '@cloudbeaver/core-root';
-import type { CbProjectsActiveEvent } from '@cloudbeaver/core-sdk';
+import { SessionEventSource, TopicEventHandler, ISessionEvent, ClientEventId, SessionEventTopic, SessionEventId } from '@cloudbeaver/core-root';
+import type { CbProjectsActiveEvent, CbProjectUpdateEvent as IProjectUpdateEvent } from '@cloudbeaver/core-sdk';
 import { isArraysEqual } from '@cloudbeaver/core-utils';
 
-export type IProjectInfoEvent = IBaseServerEvent<SessionEventId, SessionEventTopic>;
+export { IProjectUpdateEvent };
```
Comment (Wroud) [inline webapp/packages/core-projects/src/ProjectInfoEventHandler.ts]: `IProjectInfoEvent`

**Review by Wroud — CHANGES_REQUESTED** [review]

**Review by Wroud — APPROVED** [review]

**Pull request merged**